src/root/components/map/main-map.js

```diff
@@ -194,22 +196,23 @@ class MainMap extends React.Component {
       theMap.getCanvas().style.cursor = '';
     });
 
-    theMap.on('mousemove', 'district', e => {
-      const id = get(e, 'features.0.properties.OBJECTID').toString();
-      if (id && get(this.props, 'deployments.data.areas', []).find(d => d.id === id)) {
-        theMap.getCanvas().style.cursor = 'pointer';
-      } else {
-        theMap.getCanvas().style.cursor = '';
-      }
-    });
+    // theMap.on('mousemove', 'district', e => {
+    //   const id = get(e, 'features.0.properties.OBJECTID').toString();
+    //   if (id && get(this.props, 'deployments.data.areas', []).find(d => d.id === id)) {
+    //     theMap.getCanvas().style.cursor = 'pointer';
+    //   } else {
+    //     theMap.getCanvas().style.cursor = '';
+    //   }
+    // });
 
-    theMap.on('mousemove', 'icrc_admin0', e => {
+    theMap.on('mousemove', 'icrc_admin0', _debounce(e => {
       const feature = e.features.length ? e.features[0] : undefined;
-      if (feature) {
+      if (feature && feature.properties.ISO2 !== this.state.selectedFeatureISO) {
+        this.setState({ selectedFeatureISO: feature.properties.ISO2 });
         theMap.setLayoutProperty('icrc_admin0_highlight', 'visibility', 'visible');
         theMap.setFilter('icrc_admin0_highlight', ['==', 'OBJECTID', feature.properties.OBJECTID]);
       }
-    });
+    }, 80));
```

Comment:
@GregoryHorvath could you try tweaking this value - setting it to say 250, and see if there's an improvement?

Reply:
@batpad using `250` is worse for me. Played around with it for a bit, actually `80` seems to be somewhat of a sweet spot for this. I guess it might be a thing with the whole map for me and not the `debounce`.